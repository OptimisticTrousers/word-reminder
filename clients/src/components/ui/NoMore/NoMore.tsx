import CSSModules from "react-css-modules";

import styles from "./NoMore.module.css";

interface Props {
  name: string;
}

export const NoMore = CSSModules(
  function ({ name }: Props) {
    const lowerCaseName = name.toLowerCase();
    return (
      <div styleName="container">
        <p styleName="container__title">{`No more ${lowerCaseName}`}</p>
        <p styleName="container__description">{`Add more ${lowerCaseName} to see more ${lowerCaseName} in your collection.`}</p>
      </div>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "log" }
);
