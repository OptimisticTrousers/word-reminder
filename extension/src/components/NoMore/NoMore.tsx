import CSSModules from "react-css-modules";
import styles from "./NoMore.module.css";
import { FC } from "react";

interface Props {
  title: string;
  description: string;
}

const NoMore: FC<Props> = CSSModules(
  ({ title, description }) => {
    return (
      <div styleName="container">
        <p styleName="container__title">{title}</p>
        <p styleName="container__description">{description}</p>
      </div>
    );
  },
  styles,
  { allowMultiple: true, handleNotFoundStyleName: "ignore" }
);

export default NoMore;
