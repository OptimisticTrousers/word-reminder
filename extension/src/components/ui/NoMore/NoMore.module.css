.container {
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 2rem;
  background-color: var(--card-bkg-color);
  border-radius: 8px;
  border: 1px solid var(--border-color);
}

.container__title {
  color: var(--tertiary-text-color);
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.75rem;
}

.container__description {
  color: var(--tertiary-text-color);
  margin-bottom: 1rem;
}
