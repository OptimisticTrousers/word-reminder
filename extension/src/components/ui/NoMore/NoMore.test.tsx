import { render, screen } from "@testing-library/react";

import { NoMore } from "./NoMore";

describe("NoMore component", () => {
  it("renders title and description", async () => {
    const props = {
      name: "Words",
    };

    const { asFragment } = render(<NoMore {...props} />);

    const lowerCaseName = props.name.toLowerCase();
    const title = screen.getByText(`No more ${lowerCaseName}`);
    const message = screen.getByText(
      `Add more ${lowerCaseName} to see more ${lowerCaseName} in your collection.`
    );
    expect(title).toBeInTheDocument();
    expect(message).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders title and description in lower case when name is uppercase", async () => {
    const props = {
      name: "WORDS",
    };

    const { asFragment } = render(<NoMore {...props} />);

    const lowerCaseName = props.name.toLowerCase();
    const title = screen.getByText(`No more ${lowerCaseName}`);
    const message = screen.getByText(
      `Add more ${lowerCaseName} to see more ${lowerCaseName} in your collection.`
    );
    expect(title).toBeInTheDocument();
    expect(message).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
