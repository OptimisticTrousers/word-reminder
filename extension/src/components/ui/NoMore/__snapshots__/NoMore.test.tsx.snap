// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoMore component > renders title and description 1`] = `[Function]`;

exports[`NoMore component > renders title and description in lower case when name is uppercase 1`] = `[Function]`;
